### ISSUE TITLE
Multiple graphs in different threads

### ISSUE THREAD

**Issue by deadman2000** [SciSharp/TensorFlow.NET#370]
I've got this error:

```
Tensor tf.Tensor 'softmax_tensor:0' shape=(-1,2) dtype=TF_FLOAT is not an element of this graph.
   в Tensorflow.Graph._as_graph_element_locked(Object obj, Boolean allow_tensor, Boolean allow_operation)
   в Tensorflow._ElementFetchMapper..ctor(Object[] fetches, Func`2 contraction_fn)
   в Tensorflow._FetchMapper.for_fetch(Object fetch)
   в Tensorflow._FetchHandler..ctor(Graph graph, Object fetches, Dictionary`2 feeds, Action feed_handles)
   в Tensorflow.BaseSession._run(Object fetches, FeedItem[] feed_dict)
   в Tensorflow.BaseSession.run(Tensor fetche, FeedItem[] feed_dict)
```

Because _ElementFetchMapper getting **default** graph:
```
var g = ops.get_default_graph()
```

**Comment by Nucs** [comment]
We have been discussing multi-threading a lot lately, 
When initializing a model - it has to be done from a single-thread. 
You can later on use `session.run(..)` from multiple threads as `Session` class holds a reference to a graph internally.

**Comment by deadman2000** [comment]
My suggestion is deprecate `get_default_graph` and `get_default_session`

**Comment by Nucs** [comment]
We wanted that too but this will force the user to pass graph or session to every single `tf.*` call and as a result break the ability to easily port python code to C#.

**Comment by Oceania2018** [comment]
Right, when users develop a model or training model, they want to share some global variables. It’s hard to change at this moment.

**Comment by Nucs** [comment]
@deadman2000, I've added support for multithreading in PR https://github.com/SciSharp/TensorFlow.NET/pull/371
It'll be released to nuget in a day or two as version 0.11.2.
You can read more in-depth details in our wiki [here](https://github.com/SciSharp/TensorFlow.NET/wiki/Multithreading).
